src/Marten/MartenServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Marten
+{
+    public static class MartenServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Add Marten IDocumentStore, IDocumentSession, and IQuerySession service registrations
+        /// to your application with the given Postgresql connection string and Marten
+        /// defaults
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="connectionString">The connection string to your application's Postgresql database</param>
+        /// <returns></returns>
+        public static MartenConfigurationExpression AddMarten(this IServiceCollection services, string connectionString)
+        {
+            var options = new StoreOptions();
+            options.Connection(connectionString);
+            return services.AddMarten(options);
+        }
+
+        /// <summary>
+        /// Add Marten IDocumentStore, IDocumentSession, and IQuerySession service registrations
+        /// to your application using the configured StoreOptions
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="options">The Marten configuration for this application</param>
+        /// <returns></returns>
+        public static MartenConfigurationExpression AddMarten(this IServiceCollection services, StoreOptions options)
+        {
+            services.AddSingleton(options);
+            services.AddSingleton<IDocumentStore, DocumentStore>();
+
+            // This can be overridden by the expression following
+            services.AddSingleton<ISessionFactory, DefaultSessionFactory>();
```

Comment:
**SUGGESTION:** We should use `TryAddSingleton` method to allow user to override this implemntation. See more in [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.tryaddsingleton?view=dotnet-plat-ext-3.1).
```suggestion
            services.TryAddSingleton<ISessionFactory, DefaultSessionFactory>();
```

Reply:
I don't think that's necessary here Oskar.